pallets/gear/src/queue.rs

```diff
@@ -167,6 +167,9 @@ where
             T::DebugInfo::remap_id();
         }
 
+        // Create an instance of a builtin router
+        let builtin_router = T::BuiltinRouter::provide();
```

Comment:
The last one thing I would like to mention is that we need to bench this method and decrease initial_gas for block accordingly. That idea was demonstrated in the previous comment with a sample.

Reply:
Implemented.